src/cljs/rems/tree.cljs

```diff
@@ -0,0 +1,388 @@
+                    (assoc :sort-order (change-sort-order (:sort-column sorting)
+                                                          (:sort-order sorting)
+                                                          sort-column)))))))
+
+(rf/reg-sub
+ ::sorting
+ (fn [db [_ tree]]
+   (or (get-in db [::sorting (:id tree)])
+       {:sort-order (or (:default-sort-order tree)
+                        :asc)
+        :sort-column (or (:default-sort-column tree)
+                         (-> tree :columns first :key))})))
+
+(rf/reg-event-db
+ ::set-filtering
+ (fn [db [_ tree filtering]]
+   (assoc-in db [::filtering (:id tree)] filtering)))
+
+(rf/reg-sub
+ ::filtering
+ (fn [db [_ tree]]
+   (get-in db [::filtering (:id tree)])))
+
+(defn search
+  "Search field component for filtering a `rems.tree/tree` instance
+  which takes the same `tree` parameter as this component.
+
+  See `rems.tree/Tree` for the `tree` parameter schema." ; TODO schema
+  [tree]
+  ;; (s/validate Tree tree)
+  (let [filtering @(rf/subscribe [::filtering tree])
+        on-search (fn [value]
+                    (rf/dispatch [::set-filtering tree (assoc filtering :filters value)]))]
+    [search/search-field {:id (str (name (:id tree)) "-search")
+                          :on-search on-search
+                          :searching? false}]))
+
+(defn- sortable? [column]
+  (:sortable? column true))
+
+(defn- filterable? [column]
+  (:filterable? column true))
+
+(defn- display-row? [row filterable-columns search-terms]
+  (or (empty? filterable-columns) ; tree has no filtering enabled
+      (let [filterable-values (map (fn [column]
+                                     (str (get-in row [:columns-by-key (:key column) :filter-value])))
+                                   filterable-columns)]
+        (every? (fn [search-term]
+                  (some #(str/includes? % search-term) filterable-values))
+                search-terms))))
+
+(defn parse-search-terms [s]
+  (->> (re-seq #"\S+" (str s))
+       (map str/lower-case)))
+
+(rf/reg-sub
+ ::filtered-rows
+ (fn [[_ tree] _]
+   [(rf/subscribe [::flattened-rows tree])
+    (rf/subscribe [::filtering tree])])
+ (fn [[rows filtering] [_ tree]]
+   (let [search-terms (parse-search-terms (:filters filtering))
+         filterable-columns (filter filterable? (:columns tree))
+         matching-rows (filter #(display-row? % filterable-columns search-terms) rows)
+         matching-rows-set (set (mapv :key matching-rows))
+         matching-rows-set (if (:show-matching-parents? tree)
+                             (into matching-rows-set (mapcat :parents matching-rows))
+                             matching-rows-set)]
+     (mapv #(assoc % ::display-row? (contains? matching-rows-set (:key %))) ; performance optimization: hide DOM nodes instead of destroying them
+           rows))))
+
+(rf/reg-event-db
+ ::show-all-rows
+ (fn [db [_ tree]]
+   (assoc-in db [::max-rows (:id tree)] js/Number.MAX_SAFE_INTEGER)))
```

Comment:
Yes, this seems to be one of the missing features still. The old table does not render more than 100 rows by default. I think since this requires explicitly expanding a category we can accept that there should not be this same functionality. I will remove it.